### PR TITLE
[Yaml] Fix String offset cast error in Inline parser

### PR DESCRIPTION
**Pull request by romainneutron**
| Q | A |
| --- | --- |
| Branch? | 3.2/master |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | N/A |
| License | MIT |

### CODE REVIEW

File: src/Symfony/Component/Yaml/Tests/InlineTest.php [hunk 1]
```diff
@@ -654,4 +654,13 @@ public function getInvalidBinaryData()
...
         );
     }
+
+    /**
+     * @expectedException \Symfony\Component\Yaml\Exception\ParseException
+     * @expectedExceptionMessage Malformed inline YAML string {this, is not, yaml}
+     */
+    public function testStringOffsetCastError()
+    {
+        Inline::parse('{this, is not, yaml}');
```
Comment (Ener-Getick) [inline src/Symfony/Component/Yaml/Tests/InlineTest.php]: Actually this is valid and should be equivalent to `{ this: ~, is not: ~, yaml: ~ }`.

**Review by GuilhemN — CHANGES_REQUESTED** [review]

**Comment by Ener-Getick** [comment]
BTW my PR already [fixes that](https://github.com/symfony/symfony/pull/19750/files) but does a lot of stuff at the same time.

**Comment by romainneutron** [comment]
As we're already in a release candidate status for 3.2, adding new features is not possible. If you have a better fix, just let me know @Ener-Getick . This patch solves an issue in the current version

**Comment by Ener-Getick** [comment]
Then let's wait for 3.3, but i don't think making invalid something that should be valid is a solution.

File: src/Symfony/Component/Yaml/Tests/InlineTest.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Tests/InlineTest.php]: You are right. But this string wasn't even parsed before, was it? But we should nonetheless choose a better failing test case here.

**Comment by nicolas-grekas** [comment]
:+1:
at least we won't miserably fail with a PHP Warning :)
and throwing a parse error for a syntax not supported by the current parser is fine for now IMHO.

**Comment by Ener-Getick** [comment]
I reviewed your pr again and it looks like indeed that the input you test resulted at least in a warning as @nicolas-grekas said. 
So i'm 👍 for this (it should probably be merged in 2.7), it can't hurt.

**Comment by nicolas-grekas** [comment]
Checked: this is for master only, 2.7 already fails properly here, the strpos is new to 3.2.

**Comment by nicolas-grekas** [comment]
See #20388 on 2.7

**Comment by fabpot** [comment]
Thank you @romainneutron.

**Pull request merged**